### PR TITLE
fix lowering of tf.Squeeze op with -1 axis

### PR DESCRIPTION
**Pull request by wyzero**

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/lower_tf.cc [hunk 1]
```diff
@@ -100,7 +100,9 @@ class ConvertSqueezeOpDynamic : public OpRewritePattern<TF::SqueezeOp> {
       }
       llvm::SetVector<int64_t> squeeze_dims;
       for (int64_t i = 0; i < squeeze_dims_attr_size; ++i) {
-        squeeze_dims.insert(squeeze_dims_attr[i].cast<IntegerAttr>().getInt());
+        int64_t dim = squeeze_dims_attr[i].cast<IntegerAttr>().getInt();
+        if (dim == -1) dim += input_ty.getRank();
```
Comment (bddppq) [inline tao_compiler/mlir/disc/transforms/lower_tf.cc:104]: From https://www.tensorflow.org/api_docs/python/tf/squeeze#args, dim in `[-rank, rank)` are all valid.

Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/lower_tf.cc]: Done. I also add a UT for `-rank` case.

**Review by bddppq — APPROVED** [review]
LGTM

File: tao_compiler/mlir/disc/tests/tensorflow_ops/data/squeeze_d_f32_2.mlir [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module attributes {tf.versions = {bad_consumers = [], min_consumer = 0 : i32, producer = 0 : i32}} {
+  func @main(%arg0: tensor<?x?x?xf32>) -> tensor<?x?xf32> attributes {tf.entry_function = {inputs = "{{INPUTS}}", outputs = "{{OUTPUTS}}", input_placements="{{INPUT_PLACEMENTS}}", output_placements="{{OUTPUT_PLACEMENTS}}"}} {
+    %graph = tf_executor.graph {
+      %1:2 = tf_executor.island wraps "tf.Squeeze"(%arg0) {T = f32, device = "", squeeze_dims = [-1]} : (tensor<?x?x?xf32>) -> tensor<?x?xf32>
```
Comment (Orion34-lanbo) [inline tao_compiler/mlir/disc/tests/tensorflow_ops/data/squeeze_d_f32_2.mlir:4]: Nit: make the input as `tensor<?x?x1xf32>`

Comment (wyzero) [reply tao_compiler/mlir/disc/tests/tensorflow_ops/data/squeeze_d_f32_2.mlir:4]: It's the test case for fully dynamic shape. For partial dynamic shape case, we have another UT.

**Pull request merged**